src/lib.rs

```diff
@@ -764,44 +764,56 @@ pub enum TypeInner {
     BindingArray { base: Handle<Type>, size: ArraySize },
 }
 
-/// Constant value.
-#[derive(Debug, PartialEq)]
-#[cfg_attr(feature = "clone", derive(Clone))]
+#[derive(Debug, Clone, Copy, PartialOrd, PartialEq)]
 #[cfg_attr(feature = "serialize", derive(Serialize))]
 #[cfg_attr(feature = "deserialize", derive(Deserialize))]
 #[cfg_attr(feature = "arbitrary", derive(Arbitrary))]
-pub struct Constant {
-    pub name: Option<String>,
-    pub specialization: Option<u32>,
-    pub inner: ConstantInner,
+pub enum Literal {
+    F64(f64),
+    U64(u64),
+    I64(i64),
+    F32(f32),
+    U32(u32),
+    I32(i32),
+    Bool(bool),
```

Comment:
We know we're going to need full values - vectors, structs, and everything - so would it make more sense to call this `Value`? We don't need all the variants yet.

Reply:
Not sure I understand. This will only be for literals (scalars and bools).